### PR TITLE
[DNM] LootTable & Craftability

### PR DESCRIPTION
**Pull request by LetterN**
Makes railcraft somewhat playable without gamemode creative or `/give `

### CODE REVIEW

File: src/main/java/mods/railcraft/data/RailcraftBlockTagsProvider.java [hunk 1]
```diff
@@ -43,6 +43,6 @@ protected void addTags() {
 
   @Override
   public String getName() {
-    return "Railcraft Block Tags";
+    return "Railcraft Block Tagss";
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/data/RailcraftBlockTagsProvider.java]: Typo

File: src/main/java/mods/railcraft/world/item/RailcraftItems.java [hunk 2]
```diff
@@ -303,4 +303,118 @@ public ItemStack makeIcon() {
...
+  public static final RegistryObject<Item> RECEIVER_CIRCUIT =
+    ITEMS.register("receiver_circuit",
+      () -> new Item(new Item.Properties().tab(TAB)));
+
+  public static final RegistryObject<Item> SIGNAL_CIRCUIT =
+    ITEMS.register("signal_circuit",
+      () -> new Item(new Item.Properties().tab(TAB)));
+
+  //rails
+  public static final RegistryObject<Item> WOODEN_RAIL =
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/world/item/RailcraftItems.java:476]: Some of these already exist, e.g. `high_speed_flex_track` and `electric_flex_track`. (Flex tracks are tracks that can be outfitted with track kits (short for flexible tracks)).

File: src/main/java/mods/railcraft/world/item/crafting/RollingRecipeBuilder.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import net.minecraft.item.Item;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.tags.ITag;
+import net.minecraft.util.IItemProvider;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+
+public class RollingRecipeBuilder {
+  private final Item result;
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/world/item/crafting/RollingRecipeBuilder.java:27]: Put a space after the class definition (before the `result` field). (Applicable to other classes here too)

File: src/main/java/mods/railcraft/RailcraftConfig.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+      this.locomotiveLightLevel = builder
+        .comment("Change '14' to a number ranging from '0' to '15' to represent the dynamic lighting of the locomotive when Dynamic Lights mod is present.\nIf it is '0' then locomotive lightning will be disabled.")
+        .defineInRange("locomotiveLightLevel", 14, 0, 15);
+
+      builder.pop();
+    }
+  }
+
+  public static final ForgeConfigSpec clientSpec;
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/RailcraftConfig.java]: Move these fields to the top of the class.

File: src/main/java/mods/railcraft/battery/SimpleBattery.java [hunk 5]
```diff
@@ -6,7 +6,9 @@
  * Created by CovertJaguar on 1/15/2019 for Railcraft.
  *
  * @author CovertJaguar <http://www.railcraft.info>
+ * @deprecated Use the forge builtin module {@link net.minecraftforge.energy.IEnergyStorage IEnergyStorage}
  */
+@Deprecated
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/battery/SimpleBattery.java:11]: Just remove, we're still indev so doesn't matter.

File: src/main/java/mods/railcraft/RailcraftConfig.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  public static final Client CLIENT;
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/RailcraftConfig.java]: Use lower case as instances of `Client` aren't constants. Use this as a guidance for when to use all upper case:
1. All `static final` primitive types (Remember that all interface fields are inherently static final).
2. All `static final` object reference types that are never followed by "`.`" (dot).
3. All `static final` arrays that are never followed by "`[`" (opening square bracket).

**Review by josephtarbit — CHANGES_REQUESTED** [review]

File: src/main/java/mods/railcraft/world/item/RailcraftItems.java (hunk 2 above)
Comment (LetterN) [reply src/main/java/mods/railcraft/world/item/RailcraftItems.java:476]: these are crafting recipies to make said track. Rail != track

Comment (josephtarbit) [reply src/main/java/mods/railcraft/world/item/RailcraftItems.java:476]: Oh alright.

File: src/main/java/mods/railcraft/battery/SimpleBattery.java (hunk 5 above)
Comment (LetterN) [reply src/main/java/mods/railcraft/battery/SimpleBattery.java:11]: they're still being used, going to swap them on next pr's

**Pull request merged**